src/ui/map/map_scene.cpp

```diff
@@ -417,11 +423,54 @@ void MapScene::on_actionNewEvent()
 	if (result != QDialogButtonBox::Cancel)
 	{
 		m_map->events.push_back(event);
+		m_undoStack->push(new UndoEvent(event, this));
 		redrawMap();
 		emit mapChanged();
 	}
 }
 
+void MapScene::on_actionEditEvent() {
+	std::vector<lcf::rpg::Event>::iterator ev;
+	for (ev = m_map->events.begin(); ev != m_map->events.end(); ++ev) {
+		if (_index(cur_x, cur_y) == _index(ev->x, ev->y)) {
+			lcf::rpg::Event backup = *ev;
+			int result = EventDialog::edit(m_view, *ev, m_project);
+			if (result != QDialogButtonBox::Cancel) {
+				m_undoStack->push(new UndoEvent(backup, this));
+				emit mapChanged();
+			}
+			redrawMap();
+			return;
+		}
+	}
+}
+
+void MapScene::on_actionCopyEvent() {
+	event_clipboard = *getEventAt(cur_x, cur_y);
+	event_clipboard_set = true;
```

Comment:
imo that is not how that should be done. Better use the System clipboard for this.
But until this is properly figured out universally for all kinds of LCF data this is okay as a temporary workaround (but do not add this everywhere ;))